File: solutions/r/secret-handshake/1/secret-handshake.R
handshake <- function(n) {
    actions <- c(
        "wink",
        "double blink",
        "close your eyes",
        "jump"
    )
    bits <- as.integer(intToBits(n))[1:5]
    if ( tail(bits, n=1) == 1 ) {
        bits    <- bits[1:length(bits) - 1]
        bits    <- rev(bits)
        actions <- rev(actions)
    }
    on <- which( 1 == bits )
    handshake <- actions[on]
    if ( length(handshake) != 0 ) handshake
}

File: solutions/julia/secret-handshake/1/secret-handshake.jl
function secret_handshake(code)
    binary  = digits(code, base=2, pad=4)
    actions = [ "wink", "double blink", "close your eyes", "jump" ]

    for i = 1:size(actions)[1]
        if binary[i] == 0
            actions[i] = ""
        end
    end

    if size(binary) > size(actions)
        actions = reverse(actions)
    end

    actions = filter(a -> a != "", actions)

    return actions
end
